src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        T[] addArray = (T[]) list.toArray();
+        int lengthOfArray = addArray.length;
+        if (lengthOfArray == 0) {
+            return false;
+        }
+        Node<T> prev;
+        prev = last;
+        for (T elementAddArray : addArray) {
+            Node<T> inputNode = new Node<>(elementAddArray, null, prev);
+            if (prev == null) {
+                first = inputNode;
+            } else {
+                prev.next = inputNode;
+            }
+            prev = inputNode;
+        }
+        size += lengthOfArray;
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldNode = node(index);
+        T oldValue = oldNode.element;
+        oldNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T t) {
+        if (t == null) {
+            for (Node<T> i = first; i != null; i = i.next) {
+                if (i.element == null) {
+                    unlink(i);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> i = first; i != null; i = i.next) {
+                if (i.element.equals(t)) {
+                    unlink(i);
+                    return true;
+                }
+            }
+        }
```

Comment:
refactor this using one `while` loop and one `if` check

Reply:
I changed the code. I used do while but it took me 2 checks(if) because it might come null. Done

Reply:
Done)